datapipelines/src/test/java/io/customer/datapipelines/plugins/AutomaticApplicationLifecycleTrackingPluginTest.kt

```diff
@@ -0,0 +1,95 @@
+package io.customer.datapipelines.plugins
+
+import androidx.lifecycle.DefaultLifecycleObserver
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleOwner
+import com.segment.analytics.kotlin.core.Analytics
+import io.customer.commontest.extensions.assertCalledNever
+import io.customer.commontest.extensions.assertCalledOnce
+import io.customer.datapipelines.testutils.core.JUnitTest
+import io.customer.datapipelines.util.UiThreadRunner
+import io.mockk.every
+import io.mockk.just
+import io.mockk.mockk
+import io.mockk.runs
+import io.mockk.slot
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.fail
+
+class AutomaticApplicationLifecycleTrackingPluginTest : JUnitTest() {
+
+    private val mockProcessLifecycleOwner = mockk<LifecycleOwner>()
+    private val mockUiThreadRunner = mockk<UiThreadRunner>()
+    private val mockAnalytics = mockk<Analytics>()
+    private val lifecycleObserverCaptor = slot<DefaultLifecycleObserver>()
+
+    private val subject = AutomaticApplicationLifecycleTrackingPlugin(mockProcessLifecycleOwner, mockUiThreadRunner)
+
+    @BeforeEach
+    fun beforeEach() {
+        val uiThreadRunnerCaptor = slot<() -> Unit>()
+        every { mockUiThreadRunner.run(capture(uiThreadRunnerCaptor)) } answers { uiThreadRunnerCaptor.captured.invoke() }
+
+        val mockLifecycle = mockk<Lifecycle>()
+        every { mockProcessLifecycleOwner.lifecycle } returns mockLifecycle
+        every { mockLifecycle.addObserver(capture(lifecycleObserverCaptor)) } just runs
+
+        every { mockAnalytics.track(any()) } just runs
+        subject.setup(mockAnalytics)
+    }
+
+    @Test
+    fun `WHEN application lifecycle moves to CREATED THEN Application Foregrounded is NOT tracked`() {
```

Comment:
I like this format too 😢 but can we update it to follow the current naming format with `_` to keep consistency? 

We can collectively move to it, if we decide later on.